### PR TITLE
document key principles of packit

### PR DESCRIPTION
**Pull request by TomasTomecek**
This is important. This PR defines principles of packit: when in doubt, the principles should guide your next steps. Please review them and let me know what should changed, or please suggest new ones.

TODO
* [x] mege with https://github.com/packit-service/packit/blob/master/docs/source-git.md#principles
* [x] add "packit why"

### CODE REVIEW

File: content/about/index.md [hunk 1]
```diff
@@ -5,3 +5,30 @@ draft: false
...
+
+* Packit has a deprecation policy:
+  * We can mark a behaviour as deprecated.
+  * Deprecated content can be removed or changed after at least 3 minor
+    releases.
+  * Deprecated content is advertised in our
+    [changelog](https://github.com/packit-service/packit/blob/master/CHANGELOG.md),
+    printed in terminal or tracked in your issue tracker.
+
+* We do not bypass RPMs.
```
Comment (jpopelka) [inline content/about/index.md]: This needs more elaboration - I, for one, don't know what it means.

Comment (TomasTomecek) [reply content/about/index.md]: how 'bout now?

File: content/about/index.md [hunk 2]
```diff
@@ -10,4 +10,32 @@ Packit provides tooling and automation to integrate upstream open source project
...
+    [changelog](https://github.com/packit-service/packit/blob/master/CHANGELOG.md),
+    printed in terminal or tracked in your issue tracker.
+
+* We care about artifacts which Fedora supports: at the moment it's RPMs,
+  modules and container images.
+  * Our initial focus is solely on RPMs.
+
+* Packit respects Fedora guidelines.
+
+* We want latest content in Fedora Rawhide, but only if it works.
```
Comment (phracek) [inline content/about/index.md]: The sentence `only if it works` has a general meaning. 
You can think, content can be installed, content can be executed. Does it cover the tests?
I don't think, general meaning is correct. Can you please rephrase it a bit more?

File: content/about/index.md [hunk 3]
```diff
@@ -10,4 +10,32 @@ Packit provides tooling and automation to integrate upstream open source project
...
+  * We can mark a behaviour as deprecated.
+  * Deprecated content can be removed or changed after at least 3 minor
+    releases.
+  * Deprecated content is advertised in our
+    [changelog](https://github.com/packit-service/packit/blob/master/CHANGELOG.md),
+    printed in terminal or tracked in your issue tracker.
+
+* We care about artifacts which Fedora supports: at the moment it's RPMs,
+  modules and container images.
+  * Our initial focus is solely on RPMs.
```
Comment (phracek) [inline content/about/index.md]: Digging. The initial focus is RPMs, but generally, it works also in any upstream repo, including containers.

File: content/about/index.md [hunk 4]
```diff
@@ -10,4 +10,32 @@ Packit provides tooling and automation to integrate upstream open source project
 
 https://github.com/packit-service/packit
 
+
+### The key principles of packit
+
+* Reuse existing tools and services: don't reinvent the wheel.
```
Comment (lachmanfrantisek) [inline content/about/index.md]: We should make this true..;-)

Comment (TomasTomecek) [reply content/about/index.md]: ok, I'll update

File: content/about/index.md (hunk 2 above)
Comment (TomasTomecek) [reply content/about/index.md]: please suggest

File: content/about/index.md (hunk 3 above)
Comment (TomasTomecek) [reply content/about/index.md]: Yes, but the focus is RPMs: the fact that one or two commands work well with other formats is bonus points for us, but we can't build container images for example.

File: content/about/index.md (hunk 4 above)
Comment (lachmanfrantisek) [reply content/about/index.md]: I didn't expect the change in text but on our side to do it....;-)

File: content/about/index.md (hunk 2 above)
Comment (phracek) [reply content/about/index.md]: My suggestions:

`...., but only if it works, as CI tests passed, package build finishes with success by koji.`

File: docs/about/index.html [hunk 5]
```diff
@@ -74,6 +74,45 @@ <h1 class="article-title" itemprop="name">About</h1>
 <h3 id="github">Github</h3>
 
 <p><a href="https://github.com/packit-service/packit">https://github.com/packit-service/packit</a></p>
+
+<h3 id="the-key-principles-of-packit">The key principles of packit</h3>
```
Comment (dhodovsk) [inline docs/about/index.html]: Is it intended to have this section in packit home page (<hostname>/) and also in about page (<hostname>/about)?

File: content/about/index.md [hunk 6]
```diff
@@ -10,4 +10,33 @@ Packit provides tooling and automation to integrate upstream open source project
 
 https://github.com/packit-service/packit
 
+
+### The key principles of packit
```
Comment (dhodovsk) [inline content/about/index.md]: How these relate to principles in this page: https://github.com/packit-service/packit/blob/master/docs/source-git.md#principles Are the ones only about source git, or is there some intersection?

File: content/about/index.md [hunk 7]
```diff
@@ -10,4 +10,33 @@ Packit provides tooling and automation to integrate upstream open source project
...
+
+* We care about artifacts which Fedora supports: at the moment it's RPMs,
+  modules and container images.
+  * Our initial focus is solely on RPMs.
+
+* Packit respects Fedora guidelines.
+
+* We want latest content in Fedora Rawhide, but only if it works.
+
 <!--more-->
```
Comment (jpopelka) [inline content/about/index.md]: Does it make sense to have the `<!--more-->` at the very bottom of the page?

File: content/about/index.md (hunk 4 above)
Comment (TomasTomecek) [reply content/about/index.md]: and where do we reinvent?

File: docs/about/index.html (hunk 5 above)
Comment (TomasTomecek) [reply docs/about/index.html]: Yes: we should have a decent landing page (/): newcomers should learn about packit as soon as possble: I hope that we achieve that with the principles.

I also like that we have the blog posts at /, so people know what the latest release is.

I'm also planning to add the "packit why" to the about page, probably only to /about.

WDYT?

File: content/about/index.md (hunk 6 above)
Comment (TomasTomecek) [reply content/about/index.md]: Whoops, I forgot about this: we should merge them.

File: content/about/index.md (hunk 7 above)
Comment (TomasTomecek) [reply content/about/index.md]: I want the principles to be displayed on the landing page. And probably add more content to the about page.

File: content/about/index.md (hunk 4 above)
Comment (lachmanfrantisek) [reply content/about/index.md]: I've been thinking about overlap with tito/rdopkg/rpkg, but I understand that they're hard to incorporate and out of our control...

If I found anything that we can seriously use, I'll write it in my WIP research...

Comment (TomasTomecek) [reply content/about/index.md]: Yes, please!

Comment (TomasTomecek) [reply content/about/index.md]: (this is why I reworded to: "where it makes sense") :P

**Comment by TomasTomecek** [comment]
Ready for processing!

File: content/source-git.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+
+The goal of packit is to provide automation and tooling to interact with
+source-git repositories so you don't have to touch dist-git ever again. Our
+plan is to center development experience around upstream repositories and
+source-git.
+
+Upstream repositories and source-git repositories are pretty much the same
+thing. Creating source-git only makes sense when the upstream does not accept
+downstream spec file or adding spec file to such a project doesn't make sense.
+
```
Comment (jpopelka) [inline content/source-git.md]:
```suggestion
<!--more-->
```

Comment (TomasTomecek) [reply content/source-git.md]: heh, I need to regen now locally anyway :D

File: docs/about/index.html (hunk 5 above)
Comment (dhodovsk) [reply docs/about/index.html]: That sounds good :)

**Pull request merged**